packages/manager/src/managers/customTypes/CustomTypesManager.ts

```diff
@@ -167,6 +180,247 @@ export class CustomTypesManager extends BaseManager {
 		};
 	}
 
+	private updateCRCustomType(
+		args: { customType: CRCustomType } & CustomTypeFieldIdChangedMeta,
+	): { customType: CRCustomType; changed: boolean } {
+		const { previousPath, newPath } = args;
+
+		let changed = false;
+		const customType = shallowClone(args.customType);
+
+		const [previousId] = previousPath;
+		const [newId] = newPath;
+
+		if (!previousId || !newId || typeof customType === "string") {
+			return { changed, customType };
+		}
+
+		if (customType.fields) {
+			const newFields = customType.fields.map((fieldArg) => {
+				const field = shallowClone(fieldArg);
+
+				const previousId = previousPath[1];
+				const newId = newPath[1];
+
+				if (!previousId || !newId) {
+					return field;
+				}
+
+				if (typeof field === "string") {
+					if (field === previousId && field !== newId) {
+						// We have reached a field id that matches the id that was renamed,
+						// so we update it new one. The field is a string, so return the new
+						// id.
+						changed = true;
+
+						return newId;
+					}
+
+					return field;
+				}
+
+				if (field.id === previousId && field.id !== newId) {
+					// We have reached a field id that matches the id that was renamed,
+					// so we update it new one.
+					// Since field is not a string, we don't exit, as we might have
```

Comment:
I'm a bit uncertain about this statement. Even though it won't hurt to keep going, we might be ok by returning here, as I don't think we can have more than one rename in the same call by using the UI.